### PR TITLE
Dummy read to clear any pending read on SPI

### PR DESCRIPTION
**Pull request by Shetty-Anush**
## Pull Request Description

Dummy read in spi dma callback function clears appropriate data register

## PR Type
- [x] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: drivers/platform/stm32/stm32_spi.c [hunk 1]
```diff
@@ -633,7 +633,12 @@ void stm32_spi_dma_callback(struct no_os_dma_xfer_desc *old_xfer,
 	sdesc->stm32_spi_dma_done = true;
 
 	/* Dummy read to clear any pending read on SPI */
+#ifndef SPI_SR_RXNE
```
Comment (buha) [inline drivers/platform/stm32/stm32_spi.c:636]: i see that the newer chips don't have this bit in the SR register so this is your method of distinguishing SPI peripherals for newer chips like H5 from the older chips like F4

could you add a comment on this line explaining this ?

**Review by buha — COMMENTED** [review]
Also, drop the `1) ` from the commit description, a list with 1 item is not a list...

File: drivers/platform/stm32/stm32_spi.c (hunk 1 above)
Comment (Shetty-Anush) [reply drivers/platform/stm32/stm32_spi.c:636]: The file already utilizes the SPI_SR_RXNE and SPI_SR_TXE bits to distinguish between the chips. Given this consistent usage, is the comment needed?

Comment (buha) [reply drivers/platform/stm32/stm32_spi.c:636]: i just saw what Ramona wrote here:
https://github.com/analogdevicesinc/no-OS/blame/57499efc27ba9980f3856ceee1dd629062a1c4d5/drivers/platform/stm32/stm32_spi.c#L387-L390

no need anymore :D

**Review by buha — APPROVED** [review]

**Comment by buha** [comment]
Xilinx and STM32 CI builds fail, but for unrelated reasons so I'm merging.

**Pull request merged**